Analysis/include/Coda3EventDecoder.h

```diff
@@ -0,0 +1,112 @@
+		Coda3EventDecoder() : fControlEventFlag(kFALSE), TSROCNumber(0) { }
+		~Coda3EventDecoder() { }
+public:
+// Encoding Functions
+	virtual std::vector<UInt_t> EncodePHYSEventHeader();
+	virtual void EncodePrestartEventHeader(int* buffer, int buffer_size, int runnumber, int runtype = 0);
+  virtual void EncodeGoEventHeader(int* buffer, int buffer_size);
+  virtual void EncodePauseEventHeader(int* buffer, int buffer_size);
+  virtual void EncodeEndEventHeader(int* buffer, int buffer_size);
+
+public:
+// Decoding Functions
+  virtual Int_t DecodeEventIDBank(UInt_t *buffer);
+public:
+// Boolean Functions
+	virtual Bool_t IsPhysicsEvent();
+private:
+// Debugging Functions
+	void printUserEvent(const UInt_t *buffer);
+	virtual void PrintDecoderInfo(QwLog& out);
+protected:
+	UInt_t InterpretBankTag(UInt_t tag);
+	Int_t trigBankDecode(UInt_t* buffer);
+	void trigBankErrorHandler( Int_t flag );
+
+protected:
+	ULong64_t GetEvTime() const { return evt_time; }
+	void SetEvTime(ULong64_t evtime) { evt_time = evtime; }
+	UInt_t tsEvType, block_size;
+  ULong64_t evt_time; // Event time (for CODA 3.* this is a 250 Mhz clock)
+  UInt_t trigger_bits; //  (Not completely sure) The TS# trigger for the TS
+public:
+  enum { HED_OK = 0, HED_WARN = -63, HED_ERR = -127, HED_FATAL = -255 };
+  class coda_format_error : public std::runtime_error {
+  public:
+    explicit coda_format_error( const std::string& what_arg )
+      : std::runtime_error(what_arg) {}
+    explicit coda_format_error( const char* what_arg )
+      : std::runtime_error(what_arg) {}
+  };
+
+  class TBOBJ {
+  public:
+     TBOBJ() : blksize(0), tag(0), nrocs(0), len(0), tsrocLen(0), evtNum(0),
+               runInfo(0), start(nullptr), evTS(nullptr), evType(nullptr),
+               TSROC(nullptr) {}
+     void     Clear() { memset(this, 0, sizeof(*this)); }
+     uint32_t Fill( const uint32_t* evbuffer, uint32_t blkSize, uint32_t tsroc );
+     bool     withTimeStamp()   const { return (tag & 1) != 0; }
+     bool     withRunInfo()     const { return (tag & 2) != 0; }
+     bool     withTriggerBits() const { return (tsrocLen > 2*blksize);}
+
+     uint32_t blksize;          /* total number of triggers in the Bank */
+     uint16_t tag;              /* Trigger Bank Tag ID = 0xff2x */
+     uint16_t nrocs;            /* Number of ROC Banks in the Event Block (val = 1-256) */
+     uint32_t len;              /* Total Length of the Trigger Bank - including Bank header */
+     uint32_t tsrocLen;         /* Number of words in TSROC array */
+     uint64_t evtNum;           /* Starting Event # of the Block */
+     uint64_t runInfo;          /* Run Info Data (optional) */
+     const uint32_t *start;     /* Pointer to start of the Trigger Bank */
+     const uint64_t *evTS;      /* Pointer to the array of Time Stamps (optional) */
+     const uint16_t *evType;    /* Pointer to the array of Event Types */
+     const uint32_t *TSROC;     /* Pointer to Trigger Supervisor ROC segment data */
```

Comment:
The pointer to arrays of misc. TI bank data is for multiblock (and I think multi TI) events. LoadTTrigBankInfo(int index) is how you select which value to use when loading the evt_time (evTS), tsEvType (evTyp), and trigger_bits (if applicable).